main.go

```diff
@@ -157,8 +160,15 @@ func main() {
 		logrus.WithError(err).Fatal("error creating listener")
 	}
 
-	err = http.Serve(emfileAwareTcpListener{ln.(*net.TCPListener), logger}, nil)
-	if err != nil {
-		logrus.WithError(err).Fatal("error serving requests")
+	if len(*tlsCertPath) != 0 && len(*tlsKeyPath) != 0 {
+		err = http.ServeTLS(emfileAwareTcpListener{ln.(*net.TCPListener), logger}, nil, *tlsCertPath, *tlsKeyPath)
```

Comment:
Should we be using something like https://github.com/dyson/certman here? (That one seems old - just using it as an example.)

Reply:
Hmm yeah that could be a good addition.

Reply:
Is loading the cert on every request an acceptable solution? Seems like it adds fairly minimal overhead and the implementation is much simpler than using file watchers

Reply:
Huh, interesting. That's probably fine for an exporter, but I'm far from an expert in this area.

Reply:
https://github.com/digitalocean/ceph_exporter/pull/210/commits/ef01452f1cd4b7c3cf6affc8b94ff3ac974dbb4b what that looks like